src/models/attachedfiltersmodel.cpp

```diff
@@ -296,7 +293,11 @@ void AttachedFiltersModel::add(QmlMetadata* meta)
     if (m_producer->is_valid() && mlt_service_tractor_type != m_producer->type() && !QmlApplication::confirmOutputFilter()) {
         return;
     }
+    MAIN.undoStack()->push(new Filter::AddCommand(*this, meta));
+}
 
+int AttachedFiltersModel::newService(QmlMetadata* meta, Mlt::Producer& producer, int& mltIndex)
```

Comment:
When I was working on the markersmodel, I had trouble remembering which functions created commands and which ones actually performed the action. So I started a convention to add "do" prefix on functions that perform the action. Application code calls the functions without the "do" prefix and only the commands call the "do" functions. Maybe a convention like that would help in this model.

Reply:
I like this suggestion and did it,